### ISSUE TITLE
Google analytics updates

### ISSUE THREAD

**Issue by jrosindell** [OneZoom/OZtree#374]
The tacking code needs to be updated as google now has new features.  We should ensure though that the settings are still consistent with our cookie policy (e.g. it shouldn't track things at too fine a resolution to be mapped to individuals).

Connected with this, the trackers on the main tree viewer no longer work and that's been an issue for along while affecting our stats on views

**Comment by lentinj** [comment]
@jrosindell Could you update this with a list of events that should be instrumented (regardless of whether it should already work)?

The data-retention is something managed in the admin screens IIRC, so will be something you'll need to poke.

**Comment by jrosindell** [comment]
Jamie, perhaps I could give you access to the Google Analytics portal as a team member, we need updated trackers on all pages.  Also, the tags at present aggregate the data at a resolution enabling us to forego the need for explicit cookie agreement.  This is something we'd need to retain.

**Comment by lentinj** [comment]
I don't think access to the production portal is useful for me, I'd use a test GA token for testing anyway.

I think retaining the (lack of) cookie agreement is basically how the data retention is configured in the portal, but will check.

**Comment by lentinj** [comment]
https://github.com/OneZoom/OZtree/blob/main/static/site-specific/README.markdown says that the google-analytics file is empty, but it's not, it has a genuine GA property code in. Was this intentional, i.e. should all OneZoom instances? This is possibly good for museum instances, you don't have to bother configuring it, but bad for test setups.

If doing this from scratch I'd suggest:

* GA ID configuration parameter in ``appconfig.ini``, since that's a standard place for site-specific gubbins
* A template include that either adds GA tracking if there's a code set, or does nothing if not.
* Add the template include to page layout templates.

...but would this break too much to be worth doing? i.e. what do existing museum setups change?

Also, I can only find tracking for page loads, is that enough or do we need to track e.g. searches?

**Comment by jrosindell** [comment]
I think your suggestion is the right way to go about it but remember that for the tree viewer itself it's probably loading a different template.  We would want it added to all the museum display pages too - every page that gets served.  Museum displays are not often installed locally they just point to the URL (or are not updated at all) - we can always add GA to the museum display installations if this becomes an issue in future.

Searches are logged in the database already via our own tracking so don't need GA attention.

In conclusion the main thing to be sure of in implementing your suggestion is that you're using the latest tags and they the tags are on all pages and that they don't require a cookie notice.

**Comment by lentinj** [comment]
The above reworks GA support entirely as above, and now uses gtags.js.

As for switching to GA4, you have to make a new property separate from the old one:

* Add a new GA4 property
* Add a data stream to that property
* Use the GA code from the "Global site tag (gtag.js)" snippet in ``analytics.ga_code``

https://support.google.com/analytics/answer/9744165

I've done this already and created a OneZoomProd: ``G-3JSV4WMN8T``.
